Wrappers/Python/cil/optimisation/functions/TotalVariation.py

```diff
@@ -291,24 +358,21 @@ def L(self, value):
             raise TypeError('The Lipschitz constant is a real positive number')
 
     def calculate_Lipschitz(self):
+        r""" Default value for the Lipschitz constant."""
+        
         # Compute the Lipschitz parameter from the operator if possible
         # Leave it initialised to None otherwise
         self._L = (1./self.gradient.norm())**2  
     
     @property
     def gradient(self):
-        '''creates a gradient operator if not instantiated yet
-
-        There is no check that the variable _domain is changed after instantiation (should not be the case)'''
-        if self._gradient is None:
-            if self._domain is not None:
-                self._gradient = GradientOperator(self._domain, correlation = self.correlation, backend = self.backend)
+        r""" GradientOperator is created using the self._domain.
+        """
```

Comment:
The docstring is not clear. The old version was better in my opinion. Of course it has to be adapted to the new functionality, which is returning a new `GradientOperator` all the times (if `self._domain` is defined). 

I think what's missing is a line saying that this will return a configured `GradientOperator` only after a (user) call to any of `__call__` or `proximal`. The code seems to define `_domain` also with a call to `projection_P` which I don't think it's meant to be directly called by a user. So probably these lines could also be removed?

https://github.com/TomographicImaging/CIL/blob/f705ab4845fdefdfba5dae5ccfa7a68568147f49/Wrappers/Python/cil/optimisation/functions/TotalVariation.py#L226-L229

Reply:
I am not against of adding `property` for the `strong_convexity_constant`. However, we need to do this the right way and get the best from the 2 worlds math and software. IMHO, the `strong_convexity_constant` is not a TV property but it should be a `Function` class property. I previously suggested to have this during the [PDHG-strongly convex](https://github.com/TomographicImaging/CIL/pull/1030) but we did not follow that route. Maybe it's time to do it now.

Making `strong_convexity_constant` a property has exactly the same design as the `.L` property which refers to the Lipschitz constant. Atm, we do not have a lot of `Function`s and only one `L2NormSquared` is strongly convex by definition.

Reply:
In terms of type/value checking, we need to decide how we want to protect and inform our users/developers about it.
I am not sure that adding properties everywhere in our scripts is a good approach. Maybe we can rely on a good documentation and adopting [PEP484](https://peps.python.org/pep-0484/).

Reply:
- Documentation on the [gradient property](https://github.com/TomographicImaging/CIL/pull/1137/files#diff-78e9fbc2ddef346facdeca7c5fba265fbc95681cb8b0cd99ed1d41f2a0e8ee7eR361) is fixed.
- Remove `.geometry` from [projection_P](https://github.com/TomographicImaging/CIL/pull/1137/files#diff-78e9fbc2ddef346facdeca7c5fba265fbc95681cb8b0cd99ed1d41f2a0e8ee7eR219) method